blog/2025-02-18-opentofu-1-10-oci-and-supply-chain-security.md

```diff
@@ -0,0 +1,84 @@
+
+```hcl
+module "foo" {
+  source = "oci://ghcr.io/yourname/terraform-your-module"
+}
+```
+
+In contrast, providers do not support an `oci://` prefix. They are strictly tied to the format of `HOSTNAME/NAMESPACE/TYPE` (e.g. `registry.opentofu.org/hashicorp/aws`), where `HOSTNAME` defaults to `registry.opentofu.org` and `NAMESPACE` defaults to `hashicorp` for historic reasons.
+
+Not all OCI registries have an addressing scheme that clearly fits OpenTofu's provider addresses. Without a scheme component, OpenTofu must determine whether the specified hostname corresponds to an OCI registry or an OpenTofu/Terraform registry.
+
+As you can see, there is a lot remaining to be done. For now, OpenTofu will make use of its [CLI configuration](https://opentofu.org/docs/cli/config/config-file/#provider-installation). In order to use OCI for providers, you will have to create a `.tofurc` file in your home directory with a specific configuration that will tell OpenTofu which hosts are OCI registries and how to map the provider addresses to OCI addresses.
+
+## Supporting security scanners: SBOM
+
+In our survey 85% of respondents indicated that they would be interested in using a security scanner in conjunction with OpenTofu. The most popular security scanner, [Trivy](https://trivy.dev/), supports scanning Go binaries within container images out of the box, reporting known security vulnerabilities in them based on the dependency list automatically. Unfortunately, the same is not true for ZIP artifacts. While we could have chosen a standard container image layout to store providers, it would rob us of the ability to use the same provider hashes and break the lock file during migration. It would also add quite a bit of complexity to the implementation, so ultimately we ended up using the ZIP artifacts in an ORAS-style layout.
+
+Not all is lost, though. The security industry has steadily been moving towards supporting Software Bill of Materials artifacts (SBOM). These SBOMs allow a software author to publish an artifact containing a list of all dependencies and their versions. Security scanners like Trivy can use these SBOMs to perform the same license and security scans they would by inspecting the binaries. Since OCI supports attaching extra files to their artifacts, security scanners can automatically pick up these artifacts and perform the security scan that way.
+
+So all that's left to do is tie SBOMs into the OpenTofu ecosystem, which is what the [SBOM RFC](https://github.com/opentofu/opentofu/pull/2494) attempts to address. As a first step, provider authors should be able to integrate an SBOM generator like [Syft](https://github.com/anchore/syft) into their release workflow. The OpenTofu Registry will then pick up the SBOM artifacts and expose them in the registry API.
```

Comment:
"all that's left to do" seems to be doing a lot of work in this sentence :upside_down_face: